ipc/arwenpart/blockchainGateway.go

```diff
@@ -357,6 +287,67 @@ func (blockchain *BlockchainHookGateway) GetBuiltinFunctionNames() vmcommon.Func
 
 // GetAllState forwards a message to the actual hook
 func (blockchain *BlockchainHookGateway) GetAllState(address []byte) (map[string][]byte, error) {
-	//TODO implement this
-	return nil, nil
+	request := common.NewMessageBlockchainGetAllStateRequest(address)
+	rawResponse, err := blockchain.messenger.SendHookCallRequest(request)
+	if err != nil {
+		return nil, err
+	}
+
+	if rawResponse.GetKind() != common.BlockchainGetAllStateResponse {
+		return nil, common.ErrBadHookResponseFromNode
+	}
+
+	response := rawResponse.(*common.MessageBlockchainGetAllStateResponse)
+	return response.AllState, request.GetError()
+}
+
+// GetUserAccount forwards a message to the actual hook
+// TODO: Perhaps cache GetUserAccount()? Since when it is called with address == contract address, the whole code is fetched.
+func (blockchain *BlockchainHookGateway) GetUserAccount(address []byte) (vmcommon.UserAccountHandler, error) {
+	request := common.NewMessageBlockchainGetUserAccountRequest(address)
```

Comment:
you can create this cache at the level of arwen context. and clean every time

Reply:
We have some older todo's:
```
// TODO must get the code from the OutputAccount, if present
```
These will solve this issue as well. Will be done in another PR - is it ok?